packages/ethereum-group-pcd/src/EthereumGroupPCD.ts

```diff
@@ -0,0 +1,329 @@
+  );
+
+  const publicInputMsgHash = "0x" + publicInput.msgHash.toString("hex");
+
+  if (msgHash !== publicInputMsgHash) {
+    throw new Error(
+      `public input message hash ${publicInputMsgHash} does not match commitment ${message} hash ${msgHash} `
+    );
+  }
+
+  const semaphoreSignature = await SemaphoreSignaturePCDPackage.prove({
+    identity: {
+      argumentType: ArgumentTypeName.PCD,
+      pcdType: SemaphoreIdentityPCDTypeName,
+      value: args.identity.value,
+    },
+    signedMessage: {
+      argumentType: ArgumentTypeName.String,
+      value: Buffer.from(proof).toString("hex"),
+    },
+  });
+
+  return new EthereumGroupPCD(
+    uuid(),
+    {
+      publicInput: publicInput,
+      groupType:
+        args.groupType.value === "address"
+          ? GroupType.ADDRESS
+          : GroupType.PUBLICKEY,
+    },
+    {
+      signatureProof: await SemaphoreSignaturePCDPackage.serialize(
+        semaphoreSignature
+      ),
+      ethereumGroupProof: Buffer.from(proof).toString("hex"),
+    }
+  );
+}
+
+export async function verify(pcd: EthereumGroupPCD): Promise<boolean> {
+  const semaphoreSignature = await SemaphoreSignaturePCDPackage.deserialize(
+    pcd.proof.signatureProof.pcd
+  );
+  const proofValid = await SemaphoreSignaturePCDPackage.verify(
+    semaphoreSignature
+  );
+
+  // the semaphore signature of the group membership proof must be valid
+  if (!proofValid) {
+    return false;
+  }
+
+  // the string that the semaphore signature signed must equal to the ethereum
+  // signature of the commitment
+  if (semaphoreSignature.claim.signedMessage !== pcd.proof.ethereumGroupProof) {
+    return false;
+  }
+
+  const deserializedSignatureProof =
+    await SemaphoreSignaturePCDPackage.deserialize(
+      pcd.proof.signatureProof.pcd
+    );
+
+  const deserializedIdentity =
+    deserializedSignatureProof.claim.identityCommitment;
+  const message = deserializedIdentity;
+  const msgHash = ethers.utils.hashMessage(message);
+
+  const publicInputMsgHash =
+    "0x" + pcd.claim.publicInput.msgHash.toString("hex");
+
+  if (msgHash !== publicInputMsgHash) {
+    return false;
+  }
+
+  const verifier = new MembershipVerifier(
+    pcd.claim.groupType === GroupType.ADDRESS
+      ? addrMembershipConfig
+      : pubkeyMembershipConfig
+  );
+  await verifier.initWasm();
+  const verifies = await verifier.verify(
+    new Uint8Array(Buffer.from(pcd.proof.ethereumGroupProof, "hex")),
+    pcd.claim.publicInput.serialize()
+  );
+
+  if (!verifies) {
+    return false;
+  }
+
+  return true;
+}
+
+export async function serialize(
+  pcd: EthereumGroupPCD
+): Promise<SerializedPCD<EthereumGroupPCD>> {
+  return {
+    type: EthereumGroupPCDTypeName,
+    pcd: JSONBig().stringify(pcd),
+  } as SerializedPCD<EthereumGroupPCD>;
+}
+
+export async function deserialize(
+  serialized: string
+): Promise<EthereumGroupPCD> {
+  const parsed = JSONBig().parse(serialized);
```

Comment:
if you're using `{ useNativeBigInt: true }` elsewhere, might as well use it here. on the other hand it seems to just be somewhat superfluous of an option imo - we're getting away with not using it in any of the other PCD packages.